src/symbols.c

```diff
@@ -2071,102 +2071,114 @@ static TMTag *find_best_goto_tag(GeanyDocument *doc, GPtrArray *tags)
 }
 
 
+static GPtrArray *filter_tags(GPtrArray *tags, TMTag *current_tag, gboolean definition)
+{
+	const TMTagType forward_types = tm_tag_prototype_t | tm_tag_externvar_t;
+	TMTag *tmtag, *last_tag = NULL;
+	GPtrArray *filtered_tags = g_ptr_array_new();
+	guint i;
+
+	foreach_ptr_array(tmtag, i, tags)
+	{
+		if (definition && !(tmtag->type & forward_types) ||
+			!definition && (tmtag->type & forward_types))
```

Comment:
GCC suggests parentheses, and while the indention is fairly clear, I think it's still a good idea.